src/Mailer/Renderer.php

```diff
@@ -0,0 +1,462 @@
+<?php
+declare(strict_types=1);
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Mailer;
+
+use Cake\Core\Configure;
+use Cake\Http\Client\FormDataPart;
+use Cake\Utility\Hash;
+use Cake\Utility\Security;
+use Cake\Utility\Text;
+
+class Renderer
+{
+    /**
+     * Line length - no should more - RFC 2822 - 2.1.1
+     *
+     * @var int
+     */
+    public const LINE_LENGTH_SHOULD = 78;
+
+    /**
+     * Line length - no must more - RFC 2822 - 2.1.1
+     *
+     * @var int
+     */
+    public const LINE_LENGTH_MUST = 998;
+
+    /**
+     * Constant for folder name containing email templates.
+     *
+     * @var string
+     */
+    public const TEMPLATE_FOLDER = 'email';
+
+    /**
+     * The application wide charset, used to encode headers and body
+     *
+     * @var string|null
+     */
+    protected $appCharset;
+
+    /**
+     * If set, boundary to use for multipart mime messages
+     *
+     * @var string
+     */
+    protected $boundary;
+
+    /**
+     * Email instance.
+     *
+     * @var \Cake\Mailer\Email
+     */
+    protected $email;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->appCharset = Configure::read('App.encoding');
```

Comment:
Should this be a constructor argument?

Reply:
Probably. For now I had just copied from `Email`'s constructor.